src/app/core/export/export-service/export.service.ts

```diff
@@ -34,34 +35,155 @@ export class ExportService {
    * @param config (Optional) config specifying which fields should be exported
    * @returns string a valid CSV string of the input data
    */
-  createCsv(data: any[], config?: ExportColumnConfig[]): string {
-    const allFields = new Set();
+  async createCsv<T = any>(
+    data: T[],
+    config?: ExportColumnConfig[]
+  ): Promise<string> {
+    if (!config) {
+      config = this.generateExportConfigFromData(data);
+    }
+
     const exportableData = [];
+    for (const dataRow of data) {
+      const extendedExportableRows = await this.generateExportRows(
+        dataRow,
+        config
+      );
+      exportableData.push(...extendedExportableRows);
+    }
+
```

Comment:
Maybe move the `entitySortingDataAccessor` call here so its applied to all fields just before they are exported.

Reply:
Or use `entitySortingDataAccessor` whenever assigning to a `ExportRow` and ensure the return type is `string | number`